examples/plugin/README.md

```diff
@@ -0,0 +1,47 @@
+# plugin
+
+An example of a Rust app that uses Python for a plugin. A Python extension module built using PyO3 and [`maturin`](https://github.com/PyO3/maturin) is used to provide
+interface types that can be used to exchange data between Rust and Python. This also deals with how to separately test and load python modules.
+
+# Building and Testing
+## Host application
+To run the app itself, you only need to run 
+
+```shell
+cargo run
+```
+It will build the app, as well as the plugin API, then run the app, load th plugin and show it working.
+
+## Plugin API testing
+
+The plugin API is in a separate crate `plugin_api`, so you can test it separately from the main app. 
+
+To build the API only package, first install `maturin`:
+
+```shell
+pip install maturin
+```
+
+To build use `./build_extmod.sh`, simply using `maturin develop` will fail to produce a viable extension module due to the features arrangement of PyO3.
```

Comment:
Yes, that could be done too, but I always got it wrong myself while working on this. Remembering to run the correct option was super counter-intuitive to me, as muscle memory kept me from doing it. I suppose a compromise would be to specify exact command in the readme, but keep the script also.

Reply:
> I suppose a compromise would be to specify exact command in the readme, but keep the script also.

Since this is documentation and should be generally useful, I would also suggest putting the full invocation in the readme and reduce the clutter by leaving out the script. Different people use different tools to automate such tasks (shell scripts, Makefiles, Justfiles, Python scripts, etc.) and providing a single one-liner script will not materially help them IMHO.

Reply:
This has been done as suggested.